packages/js/src/import.js

```diff
@@ -0,0 +1,144 @@
+import jQuery from "jquery";
+
+import IndexingService from "./services/IndexingService";
+
+const AioseoV4 = "WPSEO_Import_AIOSEO_V4";
+
+let importButton, importForm, spinner, loading_msg, checkMark, errorMark;
+
+/**
+ * Adds Progress UI elements in the page.
+ *
+ * @returns {void}
+ */
+function addProgressElements() {
+	jQuery( spinner ).insertAfter( importButton );
+	jQuery( checkMark ).insertAfter( spinner );
+	jQuery( errorMark ).insertAfter( spinner );
+	jQuery( loading_msg ).insertAfter( spinner );
+}
+
+/**
+ * Function called when importing progress is made.
+ *
+ * @param {number} count The amount of items processed.
+ *
+ * @returns {void}
+ */
+function importingProgress( count ) {
+	console.log( "Now processed", count, "items" );
+	spinner.show();
+	loading_msg.show();
+
+	importButton.prop( "disabled", true );
+}
+
+/**
+ * Function called when importing is completed succesfully.
+ *
+ * @returns {void}
+ */
+function importingSuccess() {
+	console.log( "Done!" );
+	spinner.hide();
+	loading_msg.hide();
+	checkMark.show();
+
+	importButton.prop( "disabled", false );
+}
+
+/**
+ * Function called when importing is completed succesfully.
+ *
+ * @param {string} e The failure string.
+ *
+ * @returns {void}
+ */
+function importingFailure( e ) {
+	console.log( "Failed: " + e );
+	spinner.hide();
+	loading_msg.hide();
+	errorMark.show();
+
+	importButton.prop( "disabled", false );
+}
+
+/**
+ * Handles the import form submission and calls the new import endpoints if necessary.
+ *
+ * @param {JQuery.Event} event The submission event.
+ *
+ * @returns {void}
+ */
+function handleImportFormSubmission( event ) {
+	const dropdown = jQuery( "[name='import_external_plugin']" );
+
+	if ( dropdown.val() === AioseoV4 ) {
+		// Do not actually submit the form.
+		event.preventDefault();
+
+		const indexingService = new IndexingService( window.yoastImportData );
+
+		indexingService.index( window.yoastImportData.restApi.importing_endpoints.aioseo, importingProgress )
+			.then( () => importingSuccess() )
```

Comment:
Should be able to simply pass this function directly without wrapping it into an anonymous function.